Signal/src/ViewControllers/GifPicker/GifPickerCell.swift

```diff
@@ -0,0 +1,161 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+class GifPickerCell: UICollectionViewCell {
+    let TAG = "[GifPickerCell]"
+
+    // MARK: Properties
+
+    var imageInfo: GiphyImageInfo? {
+        didSet {
+            AssertIsOnMainThread()
+
+            ensureLoad()
+        }
+    }
+
+    var shouldLoad = false {
+        didSet {
+            AssertIsOnMainThread()
+
+            ensureLoad()
+        }
+    }
+
+    var stillAssetRequest: GiphyAssetRequest?
+    var stillAsset: GiphyAsset?
+    var fullAssetRequest: GiphyAssetRequest?
+    var fullAsset: GiphyAsset?
+    var imageView: YYAnimatedImageView?
```

Comment:
We do "progressive" loading by loading stills (jpg or gif) and "full" gifs.  This is critical on cell connections.